io/net/bonding.py

```diff
@@ -0,0 +1,293 @@
+        self.peer_ips = [self.peer_ip1]
+        for val in self.host_interfaces:
+            cmd = "ip -f inet -o addr show %s | awk '{print $4}' | cut -d /\
+                  -f1" % val
+            local_ip = process.system_output(cmd, shell=True).strip()
+            if local_ip == "":
+                self.fail("test failed because local ip can not retrieved")
+            self.host_ips.append(local_ip)
+        for val in self.peer_interfaces:
+            msg = "ip -f inet -o addr show %s | awk '{print $4}' | cut -d /\
+                  -f1" % val
+            cmd = "ssh %s@%s \"%s\"" % (self.user, self.peer_ip1, msg)
+            peer_ip = process.system_output(cmd, shell=True).strip()
+            cmd = 'echo %s | cut -d " " -f4' % peer_ip
+            peer_ip = process.system_output(cmd, shell=True).strip()
+            if peer_ip == "":
+                self.fail("test failed because peer ip can not retrieved")
+            self.peer_ips.append(peer_ip)
+        self.peer_interfaces.insert(0, self.peer_if1)
+        self.net_mask = []
+        st = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+        for val in self.host_interfaces:
+            mask = socket.inet_ntoa(fcntl.ioctl(st.fileno(), 0x891b,
+                                                struct.pack('256s',
+                                                val))[20:24]).strip('\n')
+            self.net_mask.append(mask)
+        self.bonding_slave_file = "/sys/class/net/%s/bonding/slaves"\
+                                  % self.bond_name
+
+    def bond_remove(self, arg1):
+        '''
+        bond_remove
+        '''
+        if arg1 == "local":
+            self.log.info("Bonding configuration removed on laocal")
+            self.log.info("------------------------------------------------")
+            for ifs in self.host_interfaces:
+                cmd = "ifconfig %s down" % ifs
+                if process.system(cmd, shell=True, ignore_status=True) != 0:
+                    self.log.info("unable to bring down the interface")
+                cmd = "ifconfig %s down" % ifs
+                if process.system(cmd, shell=True, ignore_status=True) != 0:
+                    self.log.info("unable to bring down the interface")
+                cmd = "echo -%s > %s" % (ifs, self.bonding_slave_file)
+                if process.system(cmd, shell=True, ignore_status=True) != 0:
+                    self.log.info("bond removing failed in local machine")
+            cmd = "echo -%s > /sys/class/net/bonding_masters" % self.bond_name
+            if process.system(cmd, shell=True, ignore_status=True) != 0:
+                self.log.info("bond removing command failed in local machine")
+            time.sleep(5)
+        else:
+            self.log.info("Bonding configuration removed on Peer machine")
+            self.log.info("------------------------------------------------")
+            cmd = ''
+            cmd += 'ifconfig %s down;' % self.bond_name
```

Comment:
/etc/sysconfig/network-scripts/ifdown is working fine in Rhel, but in ubuntu path is different(/etc/network/interfaces)